mockld/event_data.go

```diff
@@ -32,32 +32,23 @@ func (e Event) Kind() string {
 // CanonicalizedJSONString transforms the JSON formatting of the event to make comparisons
 // reliable, as follows: 1. Drop the creationDate property, since its value is unpredictable.
 // 2. Call EventUser.CanonicalizedJSONString on user properties, if any. 3. Ensure that
-// all expected properties, if nullable, do appear in the object when null instead of being
-// omitted.
+// all nullable properties are omitted if their value is null (so we can write more concise
+// test expectations).
```

Comment:
I've reversed the previous behavior of `CanonicalizedJSONString` because I realized it's a lot simpler to just drop nulls, and compare against an expected value where nulls are omitted, than to add missing nulls and compare against an expected value that has nulls.